### PR TITLE
Added HypixelHop mode to Speed module.

### PR DESCRIPTION
**Pull request by mems01**
From [this.](https://github.com/CCBlueX/LiquidBounce/issues/344)

Messy ass code first of all, and I don't know if it really bypasses on Hypixel. If someone has a premium account out there, please try it.

EDIT: I realized that this code looks fairly similar to LiquidBounce 1.8.9's hypixel speed. But seeing the way that guy did it, I believe it works differently.

### CODE REVIEW

File: src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/movement/ModuleSpeed.kt [hunk 1]
```diff
@@ -43,10 +49,36 @@ object ModuleSpeed : Module("Speed", Category.MOVEMENT) {
...
+                player.velocity.z = 0.0
+            }
+        }
     }
 
-}
+    fun getSpeedEffectAmplifier() =
+        player.getStatusEffect(StatusEffects.SPEED)!!.amplifier.plus(1)
 
+    fun getSpeed(): Float {
```
Comment (1zuna-chan) [inline src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/movement/ModuleSpeed.kt:79]: We already does have a speed function. Using mc.player.speed (not sure)

File: src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/movement/ModuleSpeed.kt [hunk 2]
```diff
@@ -43,10 +49,36 @@ object ModuleSpeed : Module("Speed", Category.MOVEMENT) {
...
+
+    private object HypixelHop : Choice("HypixelHop", modes) {
+        val playerNetworkMovementTick = handler<PlayerNetworkMovementTickEvent> { event ->
 
+            if (event.state != EventState.PRE)
+                return@handler
+
+            if (player.moving) {
+                if (player.isOnGround) {
+                    player.updatePosition(player.x, player.y + 9.1314E-4, player.z)
```
Comment (1zuna-chan) [inline src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/movement/ModuleSpeed.kt:63]: Are you sure, that this is going to bypass?

File: src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/movement/ModuleSpeed.kt (hunk 1 above)
Comment (mems01) [reply src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/movement/ModuleSpeed.kt:79]: It kinda differs I think.

File: src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/movement/ModuleSpeed.kt (hunk 2 above)
Comment (mems01) [reply src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/movement/ModuleSpeed.kt:63]: if you are talking about player.updateposition, im not quite sure. seeing from his version of hypixelhop, i don't think it's going to cause any problems? on the other hand, the person who made the issue says to use e.getY()...

**Comment by Mysels** [comment]
Dont use it. it's out dated

**Comment by mems01** [comment]
Alright.

**Pull request closed without merging**